### PR TITLE
[X86] ptest is commutable as long as only the Z flag is used.

### LINKED ISSUE #88958: x86 codegen, AVX2: missing optimization -- the compiler could swap args of commutative intrinsic `_mm256_testz_si256`
The below function has better codegen when `MASK_FIRST` is defined:
```C++
#include <immintrin.h>

bool f2(const char* p)
{
    for (const char* e = p + 256; p != e; p+=32)
    {
        const __m256i m = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(p));
        if (_mm256_testz_si256(
#ifdef MASK_FIRST
                _mm256_set1_epi8(~0x40), m
#else            
                m, _mm256_set1_epi8(~0x40)
#endif
                ))
        {
            return true;
        }
    }
    
return false;
}
```
See https://godbolt.org/z/qjqdo5x69

I want the compiler to swap arguments automatically for commutative two-args operations.

### PR DESCRIPTION
**Pull request by topperc**
Fixes #88958.

### CODE REVIEW

**Comment by topperc** [comment]
I guess this still needs SSE 4.1 coverage. @phoebewang do you want to pick this up?

**Comment by phoebewang** [comment]
> I guess this still needs SSE 4.1 coverage. @phoebewang do you want to pick this up?

Done by https://github.com/llvm/llvm-project/commit/17b86d5978af8d171fa28763a9e5eba3ce93713a, please rebase.

**Review by RKSimon — COMMENTED** [review]
No objections from me - technically we're still missing the ability to fold some loads that appear later (from stack spills etc.) but short of making a full pseudo instruction I'm not sure whether we can do it cleanly.

**Review by RKSimon — COMMENTED** [review]
Also - what about TESTPS/TESTPD?

File: llvm/lib/Target/X86/X86InstrSSE.td [hunk 1]
```diff
@@ -5737,6 +5755,13 @@ multiclass avx_bittest<bits<8> opc, string OpcodeStr, RegisterClass RC,
             Sched<[sched.Folded, sched.ReadAfterFold]>, VEX;
 }
 
+// ptest is commutable if only the Z flag is used. If the C flag is used,
```
Comment (AlexGuteniev) [inline llvm/lib/Target/X86/X86InstrSSE.td:5758]: `testp`

Comment (Alcaro) [reply llvm/lib/Target/X86/X86InstrSSE.td:5758]: I'd spell it `testps/testpd` or `testpX`

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**